### PR TITLE
remove TrackBandwidth from NetworkClient

### PR DESCRIPTION
**Pull request by danlaine**
## Why this should be merged

We don't need to expose `TrackBandwidth` on the `NetworkClient` interface.

## How this works

Move time/bandwidth tracking to `networkClient`.

## How this was tested

Existing UT

### CODE REVIEW

**Review by darioush — APPROVED** [review]

File: x/sync/network_client.go [hunk 1]
```diff
@@ -248,7 +250,10 @@ func (c *networkClient) request(
 	}
 
 	handler := newResponseHandler()
-	c.outstandingRequestHandlers[requestID] = handler
+	c.outstandingRequestHandlers[requestID] = responseHandlerAndTime{
+		handler:     handler,
+		requestTime: time.Now(),
+	}
```
Comment (StephenButtolph) [inline x/sync/network_client.go:256]: Can't we measure the bandwidth only here? This function has all of the information needed right? It measures the start time here - and then it processes either a successful or failed request for the return value.

Feels like we don't need to introduce the `responseHandlerAndTime` struct or modify `getRequestHandler`.

File: x/sync/client.go [hunk 2]
```diff
@@ -261,12 +260,9 @@ func (c *client) get(ctx context.Context, request []byte) (ids.NodeID, []byte, e
 	}
 	if err != nil {
 		c.metrics.RequestFailed()
-		c.networkClient.TrackBandwidth(nodeID, 0)
```
Comment (StephenButtolph) [inline x/sync/client.go:264]: Pretty sure this was broken previously when `RequestAny` returned an error (as we do not guarantee that `nodeID` is populated correctly)

File: x/sync/network_client.go (hunk 1 above)
Comment (danlaine) [reply x/sync/network_client.go]: done

File: x/sync/network_client.go [hunk 3]
```diff
@@ -250,9 +246,18 @@ func (c *networkClient) request(
...
 
+	var (
+		response  []byte
+		startTime = time.Now()
+	)
+	defer func() {
+		elapsed := time.Since(startTime).Seconds()
+		bandwidth := float64(len(response))/elapsed + epsilon
+		c.peers.TrackBandwidth(nodeID, bandwidth)
+	}()
```
Comment (StephenButtolph) [inline x/sync/network_client.go:257]: I don't think this `defer` honors the prior semantics - If `handler.failed` then we should be `c.peers.TrackBandwidth(nodeID, 0)` right?

Comment (danlaine) [reply x/sync/network_client.go:257]: If `handler.failed` then `len(response)` so `bandwidth` is `0`

Comment (StephenButtolph) [reply x/sync/network_client.go:257]: `+ epsilon`

**Review by StephenButtolph — APPROVED** [review]

**Pull request merged**